### PR TITLE
Issue #11266 - allow context attributes to be used to configure max form (content size / keys)

### PR DESCRIPTION
**Pull request by joakime**
**PROOF OF CONCEPT**

_This is likely incomplete and certainly not tested (yet)._

+ Allows these common configurations to be set via `WEB-INF/web.xml` without the need for a `WEB-INF/jetty-web.xml`

The configuration in the `WEB-INF/web.xml` would look like this ...

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<web-app 
   xmlns="http://xmlns.jcp.org/xml/ns/javaee" 
   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
   xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
   metadata-complete="false"
   version="3.1">

  <context-param>
    <param-name>org.eclipse.jetty.server.Request.maxFormContentSize</param-name>
    <param-value>1000000</param-value>
  </context-param>

  <context-param>
    <param-name>org.eclipse.jetty.server.Request.maxFormKeys</param-name>
    <param-value>2000</param-value>
  </context-param>
 
  <!-- ... other config ... -->
</web-app>
```

### CODE REVIEW

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java [hunk 1]
```diff
@@ -215,8 +216,8 @@ private enum ProtectedTargetType
     private String[] _vconnectors; // connector portion, matching _vhosts array
     private Logger _logger;
     private boolean _allowNullPathInfo;
-    private int _maxFormKeys = Integer.getInteger(MAX_FORM_KEYS_KEY, DEFAULT_MAX_FORM_KEYS);
-    private int _maxFormContentSize = Integer.getInteger(MAX_FORM_CONTENT_SIZE_KEY, DEFAULT_MAX_FORM_CONTENT_SIZE);
+    private int _maxFormKeys = MAX_FORM_UNSET;
```
Comment (janbartel) [inline jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java:219]: This change would prevent the use of System properties being used, which would be removal of long released behaviour.  You could just add in support for the context attributes, or we could punt this in favour of a different solution for `jetty-web.xml`.

**Review by janbartel — CHANGES_REQUESTED** [review]

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java (hunk 1 above)
Comment (joakime) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java:219]: The support for these two System properties was just moved to the doStart() location, not removed.

Comment (janbartel) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java:219]: I see it now in `doStart()`.  However, I don't think we should be doing this development in `jetty-10`,  if we're doing it, it should be in `jetty-12`.

**Review by janbartel — CHANGES_REQUESTED** [review]

File: jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java (hunk 1 above)
Comment (joakime) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java:219]: This was a possible solution for what was brought up in Issue #11274, which meant it should be done in Jetty 10 thru Jetty 12.

But the OP (@vmassol) made some good points (see https://github.com/jetty/jetty.project/issues/11266#issuecomment-1892611178) and has convinced me that the approach in this PR isn't a great solution.

I'm totally OK with this being only for Jetty 12.
Want me to recreate this for Jetty 12 only? or drop it as a concept?

Comment (janbartel) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java:219]: IMHO we should do it for 12, and also update the doco for it with the new context attributes.

Comment (joakime) [reply jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java:219]: Will do, and I'll add unit tests!

**Comment by joakime** [comment]
Closing, will be a feature introduced in Jetty 12 instead.

**Comment by vmassol** [comment]
@joakime Hello. Thanks again for working on this. Do you know if the feature has now been integrated in Jetty 12 or not yet? I checked the release notes but couldn't see it (but I could have missed it). For the context, we're still stuck with https://github.com/jetty/jetty.project/issues/11266 and any solution would work for us, even context attributes :) thx!

**Comment by vmassol** [comment]
@joakime gentle ping to know if this feature has been added in Jetty 12, see https://github.com/jetty/jetty.project/pull/11274#issuecomment-1966033971 Thx!

**Comment by joakime** [comment]
> @joakime gentle ping to know if this feature has been added in Jetty 12, see [#11274 (comment)](https://github.com/jetty/jetty.project/pull/11274#issuecomment-1966033971) Thx!

See replacement PR #11898

**Pull request closed without merging**